File: Python_sol/String/Leetcode_49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hmap = {}
        arr = []
        counter = 1
        
        # Time complexity O(N*m^2) Space complexity O(set(N) + m)
        for ele in strs:
            el = "".join(sorted(ele))
            if hmap.get(el, 0):
                arr[hmap[el]-1].append(ele)
            else:
                hmap[el] = counter
                arr.append([ele])
                counter += 1
        return arr

File: Javascript_sol/String/Leetcode_49.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
 var groupAnagrams = function(strs) {
  let hmap = {};
  let arr = [];
  let counter = 1;
  
  strs.forEach((ele) => {
      let vars = ele.split("").sort().join("");
      
      if (hmap[vars]) {
          arr[hmap[vars]-1].push(ele);
      } else {
          hmap[vars] = counter;
          arr.push([ele])
          ++counter;
      };
      
  });
  
  return arr;
};